src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java

```diff
@@ -84,6 +85,23 @@ public FakeScheduledInterviewDao() {
     return scheduledInterviewsInRange;
   }
 
+  /**
+   * Returns a list, sorted by start time, of all ScheduledInterview objects ranging from minTime to
+   * maxTime that are for the selected position and do not already have a shadow.
+   */
+  public List<ScheduledInterview> getForPositionWithoutShadowInRange(
+      Job position, Instant minTime, Instant maxTime) {
+    List<ScheduledInterview> interviewsInRange = getInRange(minTime, maxTime);
+    Set<ScheduledInterview> notValidInterviews = new HashSet<ScheduledInterview>();
+    for (ScheduledInterview interview : interviewsInRange) {
+      if (!interview.position().equals(position) || !interview.shadowId().equals("")) {
```

Comment:
shadowId().equals("") is not equivalent to not having a shadow?

Reply:
It is equivalent

Reply:
Ah I see, misread the code I guess.